packages/app-builder/src/components/Scenario/AstBuilder/RootAstBuilderNode/RootOrWithAnd.tsx

```diff
@@ -181,20 +185,25 @@ function AndOperand({
       <div className="flex h-10 items-center justify-center">
         <RemoveButton
           onClick={() => {
-            remove(path);
+            remove(treePath);
           }}
         />
       </div>
     );
-  } else if (displayReturnValues && childBooleanReturnValue) {
+  } else if (
+    displayReturnValues &&
+    childBooleanReturnValue &&
+    !errorMessages.length
+  ) {
+    const { value } = childBooleanReturnValue;
```

Comment:
I think, this should be improved :
1. use a condition independant from the "return value"
2. adapt the "return value" and return a comp conditionnaly

Something like this 
```suggestion
  } else if (
    displayReturnValues &&
    !errorMessages.length &&
    evaluation?.returnValue?.isOmitted === false
  ) {
    const value = evaluation.returnValue.value;
```

If you really want to keep the "adapter", I would prefer to split it to prevent the `undefined` check (that is IMO a bad convention). Something like :
```ts
type NonOmittedReturnValue = {
      value: ConstantType;
      isOmitted: false;
    }

function hasReturnValue(returnValue: ReturnValue): returnValue is NonOmittedReturnValue { ...}


function adaptReturnValue(returnValue: NonOmittedReturnValue) ....
```

NB: this an example of anormally cluttered code due to what you pointed out on Notion : the mix of view/edit inside the same components

Reply:
alright 👍